validator/main.py

```diff
@@ -115,43 +120,48 @@ def get_args(self) -> Dict[str, Any]:
             "model_threshold": self._kwargs.get("model_threshold", 0.5),
         }
 
-    def get_topics_ensemble(self, text: str, candidate_topics: List[str]) -> List[str]:
+    def get_topics_ensemble(self, text: str, candidate_topics: List[str]):
         applicable_topics = []
 
-        for candidate_topic in candidate_topics:
-            candidates = [candidate_topic, "other"]
-            topic, confidence = self.get_topic_zero_shot(text, candidates)
-
-            if confidence < self._model_threshold:
+        topics, scores = self.get_topic_zero_shot(text, candidate_topics)
```

Comment:
My concern w/ this approach is that we still make individual LLM calls (line 131) for every topic with a low score. As the number of topics increase, this gets pretty expensive. Can you take a look at [this PR](https://github.com/guardrails-ai/sensitive_topics/pull/3/files#diff-540acc59a982bc0e65b93e7f059f2ee471c0885170a4666217d8ad2e1743227dR129) that uses a single LLM call?